MxAnyType/CMakeLists.txt

```diff
@@ -10,42 +15,56 @@ include(GNUInstallDirs)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 # Use a general prefix for the project
-set(VARS_PREFIX ${PROJECT_NAME})
-# set(VARS_PREFIX "MxAnyType")
+# set(VARS_PREFIX ${PROJECT_NAME})
+set(VARS_PREFIX "MxAnyType")
+
+# TODO: when exporting this folder to a new repository, this will become
+#       a new project. Uncomment the project line and the alternative
+#       VARS_PREFIX.
 
 # Build the library
 # =================
 
-# Export the includes needed to who inherits this library
-# Set this up properly for handling either BUILD and INSTALL trees
-set(${VARS_PREFIX}_INCLUDE_BUILD ${CMAKE_CURRENT_SOURCE_DIR}/include)
-set(${VARS_PREFIX}_INCLUDE_INSTALL ${CMAKE_INSTALL_INCLUDEDIR}/${VARS_PREFIX})
-
 # Add the target
-add_library(${VARS_PREFIX} STATIC ${CMAKE_CURRENT_SOURCE_DIR}/include/AnyType.h
+add_library(${VARS_PREFIX} SHARED ${CMAKE_CURRENT_SOURCE_DIR}/include/AnyType.h
                                   ${CMAKE_CURRENT_SOURCE_DIR}/include/MxAnyType.h
                                   ${CMAKE_CURRENT_SOURCE_DIR}/src/MxAnyType.cpp)
 
-
-# TODO: temporary, waiting the library becomes a shared
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-    set_target_properties(${VARS_PREFIX} PROPERTIES COMPILE_FLAGS "-fPIC")
-endif()
+set_target_properties(${VARS_PREFIX} PROPERTIES
+        VERSION ${PROJECT_VERSION} # This is for the symlink of the .so
+        PUBLIC_HEADER "include/AnyType.h;include/MxAnyType.h"
+)
 
 # Find Matlab resources
 find_package(Matlab REQUIRED MX_LIBRARY)
-target_include_directories(${VARS_PREFIX} SYSTEM PUBLIC "${Matlab_INCLUDE_DIRS}")
```

Comment:
Why did you remove the `"` in `"${Matlab_INCLUDE_DIRS}"` . `"` are typically used when the variable contains directory names with spaces.

Reply:
Restored them :+1: